### PR TITLE
Filtered out DeprecationPendingWarning for matrix subclass

### PR DESCRIPTION
**Pull request by emmanuelle**
This PR should fix Travis, current failure is due to uncaught PendingDeprecationWarning raised by scipy sparse operations. Scipy will filter out these warnings in the next release, so it will be possible to remove this cumbersome warnings logics in the future (this was already mentioned in TODO.txt, therefore I didn't have to add it).

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Nice!!! These were getting annoying

**Comment by sciunto** [comment]
Must solve #3571

**Comment by emmanuelle** [comment]
Hurray, travis is finally happy :-).

File: skimage/segmentation/tests/test_random_walker.py [hunk 1]
```diff
@@ -15,8 +15,7 @@
 PYAMG_MISSING_WARNING = r'pyamg|\A\Z'
 PYAMG_OR_SCIPY_WARNING = SCIPY_RANK_WARNING + '|' + PYAMG_MISSING_WARNING
 
-if (Version(np.__version__) >= '1.15.0' and
-        Version(scipy.__version__) <= '1.1.0'):
```
Comment (hmaarrfk) [inline skimage/segmentation/tests/test_random_walker.py]: Why is this line removed? Did scipy 1.1.0 change?

Comment (hmaarrfk) [reply skimage/segmentation/tests/test_random_walker.py]: It should probably be set to < 1.2
https://scipy.github.io/devdocs/release.1.2.0.html

**Comment by jni** [comment]
@hmaarrfk well, Travis is happy...

@emmanuelle maybe you could add/edit item in to-do to point to the relevant issue and PR history here? It might be very useful when all these checks break again...

**Comment by jni** [comment]
I take it back, looking at the SciPy 1.2 release notes as suggested by @hmaarrfk, it does look like they will filter things (do a Ctrl-F for "8887"). @emmanuelle can you add the check for SciPy <1.2?

**Comment by stefanv** [comment]
+1, after integrating the `SciPy <1.2` fix.

**Comment by emmanuelle** [comment]
@jni @hmaarrfk @stefanv It does not work as shown by Travis failure for the "pip --pre" build. Indeed the warning is still raised with scipy release candidate (0.12.0rc1), I don't understand exactly why since a PR has been merged which intends to suppress such warnings, but I also checked on my machine and the warning does fire. This is why I had removed the test on scipy version until we know exactly which version of scipy solves the issue.

**Comment by emmanuelle** [comment]
@hmaarrfk yes this should be investigated but since it's annoying on the short term that Travis is red, I'll revert just the commit.

**Comment by hmaarrfk** [comment]
agreed

**Comment by jni** [comment]
Good points @emmanuelle, I didn't realise 1.2rc was out. Thanks for handling this!

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.14.x
